Sources/Imaging/Core/ImageReslice/index.js

```diff
@@ -30,6 +31,95 @@ function vtkImageReslice(publicAPI, model) {
   let indexMatrix = null;
   let optimizedTransform = null;
 
+  function getImageResliceSlabTrap(tmpPtr, inComponents, sampleCount, f) {
+    let subTmpPtr = tmpPtr;
+    let m = inComponents;
+    const n = sampleCount - 1;
+    do {
+      let result = subTmpPtr[0] * 0.5;
+      for (let k = n - 1; k !== 0; --k) {
+        subTmpPtr = tmpPtr.subarray(inComponents);
+        result += subTmpPtr[0];
+      }
+      subTmpPtr = tmpPtr.subarray(inComponents);
+      result += subTmpPtr[0] * 0.5;
+      tmpPtr.fill(result * f, 0, 1);
+      tmpPtr = tmpPtr.subarray(1);
```

Comment:
According to the C++ implementation, this part seemed necessary.
I think this part seems to require a reassign to tmpPtr.
If there are other way, I will apply them. Or it may have been applied unnecessarily.
@finetjul , Could you please check?